tendrl/node_agent/alert/utils.py

```diff
@@ -0,0 +1,122 @@
+from tendrl.commons.utils import etcd_utils
+from tendrl.node_agent.objects.node_alert_counters import NodeAlertCounters
+from tendrl.node_agent.objects.cluster_alert_counters import ClusterAlertCounters
+from tendrl.node_agent.objects.volume_alert_counters import VolumeAlertCounters
+
+def read(key):
+    result = {}
+    job = {}
+    try:
+        job = etcd_utils.read(key)
+    except EtcdKeyNotFound:
+        pass
+    except (AttributeError, EtcdException) as ex:
+        raise ex
+    if hasattr(job, 'leaves'):
+        for item in job.leaves:
+            if key == item.key:
+                result[item.key.split("/")[-1]] = item.value
+                return result
+            if item.dir is True:
+                result[item.key.split("/")[-1]] = read(item.key)
+            else:
+                result[item.key.split("/")[-1]] = item.value
+    return result
+
+
+def get_alerts():
+    alerts_arr = []
+    try:
+        alerts = read('/alerting/alerts')
+    except EtcdKeyNotFound:
+        return alerts_arr
+    except (EtcdException, AttributeError) as ex:
+        raise ex
+    for alert_id, alert in alerts.iteritems():
+        alerts_arr.append(AlertUtils().to_obj(alert))
+    return alerts_arr
+
+
+def classify_alert(alert):
+    if 'alert_catagory' in alert.tags:
+        if alert.tags['alert_catagory'] == "node":
+            NodeAlert(
+                alert_id=alert.alert_id,
+                node_id=alert.node_id,
+                time_stamp=alert.time_stamp,
+                resource=alert.resource,
+                current_value=alert.current_value,
+                tags=alert.tags,
+                alert_type=alert.alert_type,
+                severity=alert.severity,
+                significance=alert.significance,
+                ackedby=alert.ackedby,
+                acked=alert.acked,
+                ack_comment=alert.ack_comment,
+                acked_at = alert.acked_at,
+                pid=alert.pid,
+                source=alert.source,
+                delivered=alert.delivered
+            ).save()
+        elif alert.tags['alert_catagory'] == "cluster":
+            ClusterAlert(
+                alert_id=alert.alert_id,
+                node_id=alert.node_id,
+                time_stamp=alert.time_stamp,
+                resource=alert.resource,
+                current_value=alert.current_value,
+                tags=alert.tags,
+                alert_type=alert.alert_type,
+                severity=alert.severity,
+                significance=alert.significance,
+                ackedby=alert.ackedby,
+                acked=alert.acked,
+                ack_comment=alert.ack_comment,
+                acked_at = alert.acked_at,
+                pid=alert.pid,
+                source=alert.source,
+                delivered=alert.delivered
+            ).save()
+
+def update_alert_count(alert, existing_alert):
+    counter_objects = []
+    if alert.tags['alert_catagory'] == "node":
```

Comment:
Who sets this alert_catagory ?
Will the answer vary depending on source of alert like:
* Grafana
* gluster-integration
So seeing this code I am of the feeling that whoever is pushing an alert to node-agent message socket should populate this. I know, now we make sure that alert is processed into a Tendrl alert before being pushed into node-agent socket. But can you link here to how its done in code is it automatic possibly by some classification of handlers into possibly 2 categories node and cluster and the base class auto populates this as soon as a handler plugin is added to a specific category? I am just thinking about existence of a safe default for this

Reply:
As per our discussion, i will change this alert_catagory as a mandatory parameter.